### ISSUE TITLE
Importing pypose breaks torch.compile

### ISSUE THREAD

**Issue by NicolayP** [pypose/pypose#234]
### 🐛 Describe the bug

As soon as I import pypose running the compiled function `(torch.compile(model)` of a module is broken. 

This code works fine.
```python
import torch
import warnings

class MyModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.lin = torch.nn.Linear(100, 10)

    def forward(self, x):
        return torch.nn.functional.relu(self.lin(x))

mod = MyModule()
opt_mod = torch.compile(mod)
print(opt_mod(torch.randn(10, 100)))
```

But as soon as I import pypose it breaks, somehow it's related to the backward pass. I added the very nasty stacktrace

```
[/tmp/ipykernel_1286722/1297660278.py:11](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/tmp/ipykernel_1286722/1297660278.py:11): UserWarning: GPU is not NVIDIA V100, A100, or H100. Speedup numbers may be lower than expected.
  warnings.warn(
[/home/pierre/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:279](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:279): UserWarning: changing options to `torch.compile()` may require calling `torch._dynamo.reset()` to take effect
  warnings.warn(
[/home/pierre/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:303](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:303): UserWarning: Error detected in ReluBackward0. Traceback of forward call that caused the error:
  File "[/tmp/ipykernel_1286722/1297660278.py](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/tmp/ipykernel_1286722/1297660278.py)", line 22, in forward
    return torch.nn.functional.relu(self.lin(x))
 (Triggered internally at .[./torch/csrc/autograd/python_anomaly_mode.cpp:114](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/torch/csrc/autograd/python_anomaly_mode.cpp:114).)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
---------------------------------------------------------------------------
DataDependentOutputException              Traceback (most recent call last)
File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:670](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:670), in OutputGraph.call_user_compiler(self, gm)
    669 else:
--> 670     compiled_fn = compiler_fn(gm, self.fake_example_inputs())
    671 _step_logger()(logging.INFO, f"done compiler function {name}")

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/debug_utils.py:1055](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/debug_utils.py:1055), in wrap_backend_debug..debug_wrapper(gm, example_inputs, **kwargs)
   1054 else:
-> 1055     compiled_gm = compiler_fn(gm, example_inputs)
   1057 return compiled_gm

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/__init__.py:1390](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/__init__.py:1390), in _TorchCompileInductorWrapper.__call__(self, model_, inputs_)
   1388 from torch._inductor.compile_fx import compile_fx
-> 1390 return compile_fx(model_, inputs_, config_patches=self.config)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_inductor/compile_fx.py:455](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_inductor/compile_fx.py:455), in compile_fx(model_, example_inputs_, inner_compile, config_patches)
    450 with overrides.patch_functions():
    451 
    452     # TODO: can add logging before[/after](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/after) the call to create_aot_dispatcher_function
    453     # in torch._functorch[/aot_autograd.py](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/aot_autograd.py)::aot_module_simplified::aot_function_simplified::new_func
    454     # once torchdynamo is merged into pytorch
--> 455     return aot_autograd(
    456         fw_compiler=fw_compiler,
    457         bw_compiler=bw_compiler,
    458         decompositions=select_decomp_table(),
    459         partition_fn=functools.partial(
    460             min_cut_rematerialization_partition, compiler="inductor"
    461         ),
    462         keep_inference_input_mutations=True,
    463     )(model_, example_inputs_)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/backends/common.py:48](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/backends/common.py:48), in aot_autograd..compiler_fn(gm, example_inputs)
     47 with enable_aot_logging():
---> 48     cg = aot_module_simplified(gm, example_inputs, **kwargs)
     49     counters["aot_autograd"]["ok"] += 1

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2805](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2805), in aot_module_simplified(mod, args, fw_compiler, bw_compiler, partition_fn, decompositions, hasher_type, static_argnums, keep_inference_input_mutations)
   2803 full_args.extend(args)
-> 2805 compiled_fn = create_aot_dispatcher_function(
   2806     functional_call,
   2807     full_args,
   2808     aot_config,
   2809 )
   2811 # TODO: There is something deeply wrong here; compiled_fn running with
   2812 # the boxed calling convention, but aot_module_simplified somehow
   2813 # historically returned a function that was not the boxed calling
   2814 # convention.  This should get fixed...

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163), in dynamo_timed..dynamo_timed_inner..time_wrapper(*args, **kwargs)
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2498](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2498), in create_aot_dispatcher_function(flat_fn, flat_args, aot_config)
   2496 # You can put more passes here
-> 2498 compiled_fn = compiler_fn(flat_fn, fake_flat_args, aot_config)
   2500 if not hasattr(compiled_fn, "_boxed_call"):

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1713](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1713), in aot_wrapper_dedupe(flat_fn, flat_args, aot_config, compiler_fn)
   1712     if ok:
-> 1713         return compiler_fn(flat_fn, leaf_flat_args, aot_config)
   1715 # Strategy 2: Duplicate specialize.
   1716 #
   1717 # In Haskell types, suppose you have:
   (...)
   1749 #   }
   1750 #   keep_arg_mask = [True, True, False, True]

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2087](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:2087), in aot_dispatch_autograd(flat_fn, flat_args, aot_config)
   2086     flattened_joints, _ = pytree.tree_flatten(joint_inputs)
-> 2087     fx_g = make_fx(joint_forward_backward, aot_config.decompositions)(
   2088         *joint_inputs
   2089     )
   2091 # There should be *NO* mutating ops in the graph at this point.

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:714](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:714), in make_fx..wrapped(*args)
    712 with decompose(decomposition_table), fake_tensor_mode, python_dispatcher_mode, \
    713      sym_mode, proxy_mode, disable_autocast_cache(), disable_proxy_modes_tracing(enable_current=True):
--> 714     t = dispatch_trace(wrap_key(func, args, fx_tracer), tracer=fx_tracer, concrete_args=tuple(phs))
    716 # TODO: kind of a bad way to do it, should maybe figure out a better way

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209), in _TorchDynamoContext.__call__.._fn(*args, **kwargs)
    208 try:
--> 209     return fn(*args, **kwargs)
    210 finally:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:443](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:443), in dispatch_trace(root, tracer, concrete_args)
    438 def dispatch_trace(
    439         root: Union[torch.nn.Module, Callable],
    440         tracer: Tracer,
    441         concrete_args: Optional[Tuple[Any, ...]] = None,
    442 ) -> GraphModule:
--> 443     graph = tracer.trace(root, concrete_args)
    444     name = root.__class__.__name__ if isinstance(root, torch.nn.Module) else root.__name__

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209), in _TorchDynamoContext.__call__.._fn(*args, **kwargs)
    208 try:
--> 209     return fn(*args, **kwargs)
    210 finally:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py:778](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py:778), in Tracer.trace(self, root, concrete_args)
    772         _autowrap_check(
    773             patcher, module.__dict__, self._autowrap_function_ids
    774         )
    775     self.create_node(
    776         "output",
    777         "output",
--> 778         (self.create_arg(fn(*args)),),
    779         {},
    780         type_expr=fn.__annotations__.get("return", None),
    781     )
    783 self.submodule_paths = None

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py:652](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/_symbolic_trace.py:652), in Tracer.create_args_for_root..flatten_fn(*args)
    651 tree_args = pytree.tree_unflatten(list(args), in_spec)
--> 652 tree_out = root_fn(*tree_args)
    653 out_args, out_spec = pytree.tree_flatten(tree_out)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:459](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:459), in wrap_key..wrapped(*proxies)
    457     track_tensor_tree(flat_tensors, flat_proxies, constant=None, tracer=tracer)
--> 459 out = f(*tensors)
    460 out = pytree.tree_map_only(
    461     torch.Tensor,
    462     lambda t: get_proxy_slot(t, tracer, t, lambda x: x.proxy),
    463     out
    464 )

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1156](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1156), in create_forward_or_joint_functionalized..traced_joint(primals, tangents)
   1155 def traced_joint(primals, tangents):
-> 1156     return functionalized_f_helper(primals, tangents)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1108](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1108), in create_forward_or_joint_functionalized..functionalized_f_helper(primals, maybe_tangents)
   1106 try:
   1107     # Run the joint
-> 1108     f_outs = flat_fn_no_input_mutations(fn, f_primals, f_tangents, meta, keep_input_mutations)
   1109 finally:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1076](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1076), in flat_fn_no_input_mutations(fn, primals, maybe_tangents, meta, keep_input_mutations)
   1075     primals_after_cloning = primals
-> 1076 outs = flat_fn_with_synthetic_bases_expanded(fn, primals, primals_after_cloning, maybe_tangents, meta, keep_input_mutations)
   1077 return outs

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1048](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1048), in flat_fn_with_synthetic_bases_expanded(fn, primals_before_cloning, primals_after_cloning, maybe_tangents, meta, keep_input_mutations)
   1047 assert len(meta.fw_metadata.input_info) == len(primals)
-> 1048 outs = forward_or_joint(fn, primals_before_cloning, primals, maybe_tangents, meta, keep_input_mutations)
   1049 return outs

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1017](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_functorch/aot_autograd.py:1017), in forward_or_joint(fn, primals_before_cloning, primals_after_cloning, maybe_tangents, meta, keep_input_mutations)
   1016     with fx_traceback.preserve_node_meta():
-> 1017         backward_out = torch.autograd.grad(
   1018             needed_outs,
   1019             grad_primals,
   1020             grad_outputs=needed_tangents,
   1021             allow_unused=True,
   1022         )
   1023 backward_out_iter = iter(backward_out)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:269](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:269), in grad(outputs, inputs, grad_outputs, retain_graph, create_graph, only_inputs, allow_unused, is_grads_batched)
    268 if has_torch_function(overridable_args):
--> 269     return handle_torch_function(
    270         grad,
    271         overridable_args,
    272         t_outputs,
    273         t_inputs,
    274         grad_outputs=grad_outputs,
    275         retain_graph=retain_graph,
    276         create_graph=create_graph,
    277         only_inputs=only_inputs,
    278         allow_unused=allow_unused,
    279         is_grads_batched=is_grads_batched,
    280     )
    282 if not only_inputs:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/overrides.py:1534](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/overrides.py:1534), in handle_torch_function(public_api, relevant_args, *args, **kwargs)
   1533 with _pop_mode_temporarily() as mode:
-> 1534     result = mode.__torch_function__(public_api, types, args, kwargs)
   1535 if result is not NotImplemented:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_inductor/overrides.py:38](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_inductor/overrides.py:38), in AutogradMonkeypatch.__torch_function__(self, func, types, args, kwargs)
     37     return replacements[func](*args, **kwargs)
---> 38 return func(*args, **kwargs)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:303](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/autograd/__init__.py:303), in grad(outputs, inputs, grad_outputs, retain_graph, create_graph, only_inputs, allow_unused, is_grads_batched)
    302 else:
--> 303     return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
    304         t_outputs, grad_outputs_, retain_graph, create_graph, t_inputs,
    305         allow_unused, accumulate_grad=False)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/utils/_stats.py:20](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/utils/_stats.py:20), in count..wrapper(*args, **kwargs)
     19 simple_call_counter[fn.__qualname__] = simple_call_counter[fn.__qualname__] + 1
---> 20 return fn(*args, **kwargs)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:487](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:487), in ProxyTorchDispatchMode.__torch_dispatch__(self, func, types, args, kwargs)
    486 with self.sym_mode.enable(False):
--> 487     return self.inner_torch_dispatch(func, types, args, kwargs)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:512](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:512), in ProxyTorchDispatchMode.inner_torch_dispatch(self, func, types, args, kwargs)
    510     return func(*args, **kwargs)
--> 512 out = proxy_call(self, func, args, kwargs)
    513 return out

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:345](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/fx/experimental/proxy_tensor.py:345), in proxy_call(proxy_mode, func, args, kwargs)
    343         args[0].proxy = proxy_out
--> 345 out = func(*args, **kwargs)
    347 # In some circumstances, we will be tracing in a situation where a tensor
    348 # is *statically* known to be a constant (currently, this only happens if
    349 # you run torch.tensor; deterministic factory functions like torch.arange
   (...)
    366 # propagating const-ness.  Similarly, we don't require the constant to
    367 # live on CPU, but we could.

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_ops.py:287](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_ops.py:287), in OpOverload.__call__(self, *args, **kwargs)
    286 def __call__(self, *args, **kwargs):
--> 287     return self._op(*args, **kwargs or {})

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/utils/_stats.py:20](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/utils/_stats.py:20), in count..wrapper(*args, **kwargs)
     19 simple_call_counter[fn.__qualname__] = simple_call_counter[fn.__qualname__] + 1
---> 20 return fn(*args, **kwargs)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:987](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:987), in FakeTensorMode.__torch_dispatch__(self, func, types, args, kwargs)
    986 try:
--> 987     return self.dispatch(func, types, args, kwargs)
    988 except TypeError:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:1162](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:1162), in FakeTensorMode.dispatch(self, func, types, args, kwargs)
   1161 if run_impl_check(func):
-> 1162     op_impl_out = op_impl(self, func, *args, **kwargs)
   1163     if op_impl_out != NotImplemented:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:410](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_subclasses/fake_tensor.py:410), in local_scalar_dense(fake_mode, func, arg)
    408 if fake_mode.shape_env is None:
    409     # Without symints[/symfloats](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/symfloats), cannot handle this
--> 410     raise DataDependentOutputException(func)
    411 if is_float_dtype(arg.dtype):

DataDependentOutputException: aten._local_scalar_dense.default

The above exception was the direct cause of the following exception:

BackendCompilerFailed                     Traceback (most recent call last)
Cell In[11], line 26
     24 mod = MyModule()
     25 opt_mod = torch.compile(mod)
---> 26 print(opt_mod(torch.randn(10, 100)))

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/nn/modules/module.py:1501](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/nn/modules/module.py:1501), in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:82](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:82), in OptimizedModule.forward(self, *args, **kwargs)
     81 def forward(self, *args, **kwargs):
---> 82     return self.dynamo_ctx(self._orig_mod.forward)(*args, **kwargs)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:209), in _TorchDynamoContext.__call__.._fn(*args, **kwargs)
    207 dynamic_ctx.__enter__()
    208 try:
--> 209     return fn(*args, **kwargs)
    210 finally:
    211     set_eval_frame(prior)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:337](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/eval_frame.py:337), in catch_errors_wrapper..catch_errors(frame, cache_size)
    334             return hijacked_callback(frame, cache_size, hooks)
    336 with compile_lock:
--> 337     return callback(frame, cache_size, hooks)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:404](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:404), in convert_frame.._convert_frame(frame, cache_size, hooks)
    402 counters["frames"]["total"] += 1
    403 try:
--> 404     result = inner_convert(frame, cache_size, hooks)
    405     counters["frames"]["ok"] += 1
    406     return result

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:104](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:104), in wrap_convert_context.._fn(*args, **kwargs)
    102 torch.fx.graph_module._forward_from_src = fx_forward_from_src_skip_result
    103 try:
--> 104     return fn(*args, **kwargs)
    105 finally:
    106     torch._C._set_grad_enabled(prior_grad_mode)

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:262](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:262), in convert_frame_assert.._convert_frame_assert(frame, cache_size, hooks)
    259 global initial_grad_state
    260 initial_grad_state = torch.is_grad_enabled()
--> 262 return _compile(
    263     frame.f_code,
    264     frame.f_globals,
    265     frame.f_locals,
    266     frame.f_builtins,
    267     compiler_fn,
    268     one_graph,
    269     export,
    270     hooks,
    271     frame,
    272 )

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163), in dynamo_timed..dynamo_timed_inner..time_wrapper(*args, **kwargs)
    161     compilation_metrics[key] = []
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0
    165 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:324](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:324), in _compile(code, globals, locals, builtins, compiler_fn, one_graph, export, hooks, frame)
    322 for attempt in itertools.count():
    323     try:
--> 324         out_code = transform_code_object(code, transform)
    325         orig_code_map[out_code] = code
    326         break

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py:445](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/bytecode_transformation.py:445), in transform_code_object(code, transformations, safe)
    442 instructions = cleaned_instructions(code, safe)
    443 propagate_line_nums(instructions)
--> 445 transformations(instructions, code_options)
    446 return clean_and_assemble_instructions(instructions, keys, code_options)[1]

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:311](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/convert_frame.py:311), in _compile..transform(instructions, code_options)
    298 nonlocal output
    299 tracer = InstructionTranslator(
    300     instructions,
    301     code,
   (...)
    309     mutated_closure_cell_contents,
    310 )
--> 311 tracer.run()
    312 output = tracer.output
    313 assert output is not None

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:1726](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:1726), in InstructionTranslator.run(self)
   1724 def run(self):
   1725     _step_logger()(logging.INFO, f"torchdynamo start tracing {self.f_code.co_name}")
-> 1726     super().run()

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:576](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:576), in InstructionTranslatorBase.run(self)
    571 try:
    572     self.output.push_tx(self)
    573     while (
    574         self.instruction_pointer is not None
    575         and not self.output.should_exit
--> 576         and self.step()
    577     ):
    578         pass
    579 except BackendCompilerFailed:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:540](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:540), in InstructionTranslatorBase.step(self)
    538     if not hasattr(self, inst.opname):
    539         unimplemented(f"missing: {inst.opname}")
--> 540     getattr(self, inst.opname)(inst)
    542     return inst.opname != "RETURN_VALUE"
    543 except BackendCompilerFailed:

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:1792](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/symbolic_convert.py:1792), in InstructionTranslator.RETURN_VALUE(self, inst)
   1787 _step_logger()(
   1788     logging.INFO,
   1789     f"torchdynamo done tracing {self.f_code.co_name} (RETURN_VALUE)",
   1790 )
   1791 log.debug("RETURN_VALUE triggered compile")
-> 1792 self.output.compile_subgraph(
   1793     self, reason=GraphCompileReason("return_value", [self.frame_summary()])
   1794 )
   1795 self.output.add_output_instructions([create_instruction("RETURN_VALUE")])

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:517](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:517), in OutputGraph.compile_subgraph(self, tx, partial_convert, reason)
    503     self.add_output_instructions(random_calls_instructions)
    505 if (
    506     stack_values
    507     and all(
   (...)
    514 
    515     # optimization to generate better code in a common case
    516     self.add_output_instructions(
--> 517         self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
    518         + [create_instruction("UNPACK_SEQUENCE", len(stack_values))]
    519     )
    520 else:
    521     graph_output_var = self.new_var("graph_out")

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:588](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:588), in OutputGraph.compile_and_call_fx_graph(self, tx, rv, root)
    586 assert_no_fake_params_or_buffers(gm)
    587 with tracing(self.tracing_context):
--> 588     compiled_fn = self.call_user_compiler(gm)
    589 compiled_fn = disable(compiled_fn)
    591 counters["stats"]["unique_graphs"] += 1

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/utils.py:163), in dynamo_timed..dynamo_timed_inner..time_wrapper(*args, **kwargs)
    161     compilation_metrics[key] = []
    162 t0 = time.time()
--> 163 r = func(*args, **kwargs)
    164 time_spent = time.time() - t0
    165 # print(f"Dynamo timer: key={key}, latency={latency:.2f} sec")

File [~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:675](https://vscode-remote+ssh-002dremote-002bpro-002eeps-002ehw-002eac-002euk.vscode-resource.vscode-cdn.net/home/pierre/workspace/rnn_auv/scripts/~/anaconda3/envs/torch2/lib/python3.10/site-packages/torch/_dynamo/output_graph.py:675), in OutputGraph.call_user_compiler(self, gm)
    673 except Exception as e:
    674     compiled_fn = gm.forward
--> 675     raise BackendCompilerFailed(self.compiler_fn, e) from e
    676 return compiled_fn

BackendCompilerFailed: debug_wrapper raised DataDependentOutputException: aten._local_scalar_dense.default

Set torch._dynamo.config.verbose=True for more information


You can suppress this exception and fall back to eager by setting:
    torch._dynamo.config.suppress_errors = True
```

### Versions

Collecting environment information...
PyTorch version: 2.0.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.10.11 (main, Apr 20 2023, 19:02:41) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.17.5-76051705-generic-x86_64-with-glibc2.31
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: NVIDIA GeForce GTX TITAN X
GPU 1: NVIDIA GeForce GTX TITAN X
GPU 2: NVIDIA TITAN Xp COLLECTORS EDITION
GPU 3: NVIDIA GeForce GTX TITAN X
GPU 4: NVIDIA TITAN Xp
GPU 5: NVIDIA TITAN Xp COLLECTORS EDITION
GPU 6: NVIDIA TITAN Xp COLLECTORS EDITION
GPU 7: NVIDIA TITAN Xp

Nvidia driver version: 525.85.12
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] torch==2.0.0
[pip3] torchvision==0.15.1
[conda] numpy                     1.24.3                  pypi_0    pypi
[conda] torch                        2.0.0                    pypi_0    pypi    
[conda] torchvision             0.15.1                  pypi_0    pypi

**Comment by NicolayP** [comment]
Hi @wang-chen I run it from the main branch

**Comment by zitongzhan** [comment]
Tested with the main branch but haven't exposed the issue
![image](https://user-images.githubusercontent.com/24406547/236043999-0f09fabe-b86a-4259-905b-c6afc6bbfcab.png)

**Comment by wang-chen** [comment]
@NicolayP Did you solve this problem? will close this issue if yes.
